extension/api.js

```diff
@@ -0,0 +1,80 @@
+/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "skeleton" }]*/
+/* global ExtensionAPI */
+
+const PREF_HOTFIX_DONE = "extensions.hotfix-intermediate-2018.done";
+const PREF_MASTER_PASSWORD_REQUEST = "extensions.hotfix-intermediate-2018.requestMasterPassword";
+
+this.hotfixIntermediateCert = class extends ExtensionAPI {
+  onStartup() {
+    if (Services.prefs.getBoolPref(PREF_HOTFIX_DONE, false)) {
+      // Return earlier if the hotfix has been already applied.
+      return;
+    }
+    this.applyHotFix();
+  }
+
+  async applyHotFix() {
+    let XPIDatabase;
+
+    try {
+      XPIDatabase = ChromeUtils.importESModule("resource://gre/modules/addons/XPIDatabase.sys.mjs").XPIDatabase;
+    } catch (err) {
+      console.warn("Unable to load XPIDatabase.sys.mjs", err);
+      return;
+    }
+
+    if (!XPIDatabase.verifySignatures) {
+      // If this is Firefox <63, verifySignatures was an XPIProvider method, but we are not targeting
+      // Firefox versions as old as that anyway.
+      console.warn("XPIDatabase.verifySignatures not found (Firefox version is likely too old)");
+      return;
+    }
+
+    // First, check if there's a master password on the token DB. If so,
+    // the user will need to login before we can add a certificate.
+    try {
+      let tokendb = Cc["@mozilla.org/security/pk11tokendb;1"].getService(Ci.nsIPK11TokenDB);
+      let token = tokendb.getInternalKeyToken();
+
+      if (!token.checkPassword("")) {
+        // Calling `token.login(true)` here would trigger a dialog that would be requesting
+        // the user to fill in their Master Password, that would be surprising and so we
+        // bail only do that if the user has set a pref to explicitly allow that (e.g. in
```

Comment:
something like this maybex?

```suggestion
        // bail out and only do that if the user has set a pref to explicitly allow that (e.g. in
```